server/lib/nl/detection/context.py

```diff
@@ -100,12 +100,17 @@ def merge_with_context(uttr: nl_uttr.Utterance, default_place: Place = None):
   main_vars, cmp_vars = _detect_vars(
       uttr, query_type == nl_uttr.QueryType.CORRELATION_ACROSS_VARS)
 
+  # 6. Detect entities leveraging context
+  entities = _detect_entities(uttr)
+
+  # 7. Detect properties leveraging context
+  properties = _detect_props(uttr)
+
   # 6. Populate the returned dict
   data_dict.update({
       Params.ENTITIES.value:
-          places,
-      Params.VARS.value:
-          main_vars[:_MAX_RETURNED_VARS],
+          places + entities,
```

Comment:
hmm, why are we merging them?  Should we keep them separate so "fulfill" can do the right thing?

Reply:
I had originally thought to merge them because I know we eventually want to move towards only having entities and only having variables & right now, it looks like the insight context is just used for generating place + topic urls on the front end. But maybe makes more sense to separate them for now and then merge later when we do want to merge entities + places, and variable + properties. Updated